dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,784 @@
+    peak_type = config.Property(proptype=str, default="max_amp")
+
+    def setup(self):
+        """Open connection to data index database.
+
+        Creates table if it does not exist.
+        Further, it adds the data metric.
+        """
+        # Check for database
+
+        db_file = os.path.join(self.state_dir, DB_FILE)
+        db_types = sqlite3.PARSE_DECLTYPES | sqlite3.PARSE_COLNAMES
+        self.data_index = sqlite3.connect(
+            db_file, detect_types=db_types, check_same_thread=False
+        )
+
+        cursor = self.data_index.cursor()
+        cursor.execute(CREATE_DB_TABLE)
+        self.data_index.commit()
+
+        # initalise run metric
+        self.flux_metric = self.add_data_metric(
+            "flux", labelnames=["frequency", "source", "pol"], unit="Jansky"
+        )
+
+    def run(self):
+        """Run the analyzer.
+
+         "Load stacked visibilities and beam form to the
+          location of the brightest sources."
+
+        Write fluxes for a given time and frequency to disk.
+        """
+        lat = np.radians(ephemeris.CHIMELATITUDE)
+        err_msg = ""
+
+        # Create transit tracker
+        for src in self.source_transits:
+            try:
+                src_body = fluxcat.FluxCatalog[src].skyfield
+            except KeyError as e:
+                raise DiasConfigError("Invalid source: {}".format(e))
+
+            self.logger.info(
+                "Initializing offline point source processing for {}.".format(src)
+            )
+
+            # Query files from now to period hours back
+            query_stop_time = datetime.utcnow() - self.lag
+            query_start_time = query_stop_time - self.period
+
+            # Refresh the database
+            self.refresh_data_index()
+
+            cursor = self.data_index.cursor()
+            query = "SELECT stop FROM files ORDER BY stop DESC LIMIT 1"
+            results = list(cursor.execute(query))
+            if results:
+                query_start_time = results[0][0]
+            else:
+                query_start_time = query_stop_time - self.period
+
+            self.logger.info(
+                "Searching for transits from %s to %s"
+                % (str(query_start_time), str(query_stop_time))
+            )
+
+            # nsigma distance in degree from transit from peak
+            time_delta = (
+                2
+                * self.nsigma
+                * cal_utils.guess_fwhm(
+                    400.0, pol="X", dec=src_body.dec.radians, sigma=True, seconds=True
+                )
+            )
+
+            # Find all calibration files that have transit of given source
+            f = self.Finder()
+            f.set_time_range(query_start_time, query_stop_time)
+            f.accept_all_global_flags()
+            f.only_corr()
+            f.filter_acqs((data_index.ArchiveInst.name == self.acq_suffix))
+            f.include_transits(src_body, time_delta=time_delta)
```

Comment:
Either way, I think it is important to include a flag in the attributes of the output file (currently the `is_daytime` attribute) indicating whether or not this particular source transit occurred during the night, day, or solar transit, so that people can easily filter results based on this when viewing in theremin or offline.